src/acl-check.js

```diff
@@ -44,15 +44,45 @@ function accessDenied (kb, doc, directory, aclDoc, agent, modesRequired, origin,
   return ok
 }
 
+async function getTrustedModesForOrigin (kb, agent, origin) {
+  if (!kb || !agent || !origin) {
+    return Promise.resolve([])
+  }
+  const result = await query(`
+  SELECT ?mode WHERE {
+    ${agent} ${ACL('trustedApp')} ?trustedOrigin.
+    ?trustedOrigin  ${ACL('origin')} ${origin};
+                    ${ACL('mode')} ?mode .
+  }`, kb)
+  const trustedModes = result.map(result => result['?mode'])
+  return Promise.resolve(trustedModes)
+}
+
+async function query (queryString, store) {
```

Comment:
It seems to me like the query function here has some generic functionality that should have lived elsewhere?